src/Compiler/Microsoft.CodeAnalysis.Razor/src/ComponentTagHelperDescriptorProvider.cs

```diff
@@ -299,19 +312,21 @@ static bool VerifyGetAwaiter(IMethodSymbol getAwaiter)
                     return false;
                 }
 
-                if (!returnType.GetMembers().OfType<IPropertySymbol>().Any(p => p.Name == WellKnownMemberNames.IsCompleted && p.Type.SpecialType == SpecialType.System_Boolean && p.GetMethod != null))
+                var members = returnType.GetMembers();
+
+                if (!members.OfType<IPropertySymbol>().Any(static p => p.Name == WellKnownMemberNames.IsCompleted && p.Type.SpecialType == SpecialType.System_Boolean && p.GetMethod != null))
                 {
                     return false;
                 }
 
-                var methods = returnType.GetMembers().OfType<IMethodSymbol>();
+                var methods = members.OfType<IMethodSymbol>();
 
-                if (!methods.Any(x => x.Name == WellKnownMemberNames.OnCompleted && x.ReturnsVoid && x.Parameters.Length == 1 && x.Parameters.First().Type.TypeKind == TypeKind.Delegate))
+                if (!methods.Any(static x => x.Name == WellKnownMemberNames.OnCompleted && x.ReturnsVoid && x.Parameters.Length == 1 && x.Parameters.First().Type.TypeKind == TypeKind.Delegate))
                 {
                     return false;
                 }
```

Comment:
Any consideration on making this a single foreach loop, rather than two loops?

Reply:
Good idea. I'll do that.